velox/connectors/hive/tests/FileHandleTest.cpp

```diff
@@ -38,13 +38,80 @@ TEST(FileHandleTest, localFile) {
 
   FileHandleFactory factory(
       std::make_unique<
-          SimpleLRUCache<std::string, std::shared_ptr<FileHandle>>>(1000),
+          SimpleLRUCache<FileHandleKey, std::shared_ptr<FileHandle>>>(1000),
       std::make_unique<FileHandleGenerator>());
-  auto fileHandle = factory.generate(filename).second;
+  auto fileHandle = factory.generate(FileHandleKey(filename)).second;
   ASSERT_EQ(fileHandle->file->size(), 3);
   char buffer[3];
   ASSERT_EQ(fileHandle->file->pread(0, 3, &buffer), "foo");
 
   // Clean up
   remove(filename.c_str());
 }
+
+TEST(FileHandleTest, localFileWithProperties) {
+  filesystems::registerLocalFileSystem();
+
+  auto tempFile = exec::test::TempFilePath::create();
+  const auto& filename = tempFile->getPath();
+  remove(filename.c_str());
+
+  {
+    LocalWriteFile writeFile(filename);
+    writeFile.append("foo");
+  }
+
+  FileHandleFactory factory(
+      std::make_unique<
+          SimpleLRUCache<FileHandleKey, std::shared_ptr<FileHandle>>>(1000),
+      std::make_unique<FileHandleGenerator>());
+  auto fileHandle =
+      factory
+          .generate(FileHandleKey(
+              tempFile->fileSize(), tempFile->fileModifiedTime(), filename))
+          .second;
+  ASSERT_EQ(fileHandle->file->size(), 3);
+  char buffer[3];
+  ASSERT_EQ(fileHandle->file->pread(0, 3, &buffer), "foo");
+
+  // Clean up
+  remove(filename.c_str());
+}
+
+TEST(FileHandleTest, retrieveCachedLocalFile) {
+  filesystems::registerLocalFileSystem();
+
+  auto tempFile = exec::test::TempFilePath::create();
+  const auto& filename = tempFile->getPath();
+  remove(filename.c_str());
```

Comment:
Do we need to delete and write below for this test? I am thinking an empty file will do too.